packages/Icon/src/createIconComponent.js

```diff
@@ -0,0 +1,59 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+
+export default glyphs => {
+  return class Icon extends PureComponent {
+    static displayName = 'Icon';
+    static propTypes = {
+      glyph: PropTypes.oneOf(Object.keys(glyphs)).isRequired,
+      size: PropTypes.oneOfType([
+        PropTypes.oneOf(['default', 'large', 'xlarge']),
+        PropTypes.number,
+      ]),
+    };
+
+    static defaultProps = {
+      size: 'default',
+    };
+
+    get humanReadableTitle() {
```

Comment:
Neat, but not necessary. Can we stick with functions as that is what people reading the code expect to see?

Reply:
This can also just be a plain function that is written in the module.

Ditto the other function.

Reply:
Fair enough – I rather like this (especially for the other getter), but I get your point that it may not be clear to those reading that this is where the class property is coming from (though wouldn't they look for `humanReadableTitle` if they needed to find it either way?).

I can make these either a class method, or do this in the render method once we decide on class vs functional component in here.

Reply:
When I see a value of `this.something` I assume it's an instance property, not a getter. Which may just be me, but I think that's the default way people read classes.

Additionally having it as an instance getter can make people believe that it relies on state when in actuality it's a pure function with no reliance on state.

I think having this as either a static method, directly in render, or as a standalone function in the module (unexported) are all fine.

Reply:
They aren't pure, since they do have a reliance on props, but point taken.

I'll update them to not be getters.

Reply:
So, I updated this, but didn't do a static method, standalone function, or directly in the render method.

I went with an instance method, since I wanted to avoid passing props around too much, and wanted to keep the render method more minimal.

If you're still not cool with this, I'll refactor this to return a functional component instead, and host all the logic there.